docs/nidigital/class.rst

```diff
@@ -2194,7 +2194,7 @@ unload_all_patterns
 
     .. py:currentmodule:: nidigital.Session
 
-    .. py:method:: unload_all_patterns(unload_keep_alive_pattern)
+    .. py:method:: unload_all_patterns(unload_keep_alive_pattern=False)
```

Comment:
The default contradicts the name of the function "unload all" but implicitly "not one of them".

Reply:
Agreed, this is tricky function to design. Users typically (90%) don't use `keep_alive` pattern, so it makes sense to not require them to pass any parameter by using a default value. For those that do use `keep_alive` pattern, they typically don't want to unload `keep_alive` pattern; hence the default of `False`.

We could go for a different design like renaming the method or changing its parameters, but that would make it inconsistent with LV and C API. Even though it's semantically wrong, it doesn't lead the users to do the wrong thing. I'd say we keep it.

Reply:
Ok, the first part makes sense.

> but that would make it inconsistent with LV and C API

Are you using defaults in your C API?
Per our [API design guidelines](https://github.com/ni/nimi-python/wiki/API-design-guidelines#default-values-for-methods):

>For the most part, defaults are a subset of the defaults in the corresponding LabVIEW VIs
Less defaults than in LabVIEW APIs, where we may have overdone it
Defaults can always be added later
But never removed

So the fact that it is defaulted in LabVIEW doesn't suggest strongly that we have to.

In any case, I'm ok with the default given your explanation.